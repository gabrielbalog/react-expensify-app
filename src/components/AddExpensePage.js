import React from 'react';

const AddExpensePage = () => (
	<div>
		This is from add expense Page
	</div>
);

export default AddExpensePage;