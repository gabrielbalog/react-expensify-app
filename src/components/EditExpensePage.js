import React from 'react';

const EditExpensePage = () => (
	<div>
		This is from edit expense Page
	</div>
);

export default EditExpensePage;