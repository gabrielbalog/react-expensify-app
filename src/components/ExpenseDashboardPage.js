import React from 'react';

const ExpenseDashboardPage = () => (
	<div>
		This is from dashboard Page
	</div>
);

export default ExpenseDashboardPage;