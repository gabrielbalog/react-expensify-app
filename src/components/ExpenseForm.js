import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css'

// const Date = new Date();
const now = moment();
console.log(now.format('MMM Do, YYYY'));


export default class ExpenseForm extends React.Component {
	state = {
		description: '',
		note: '',
		amount: '',
		createdAt: moment(),
		calendarFocused: false,
		error: ''
	}
	onDescriptionChange = (e) => {
		const description = e.target.value;
		this.setState(() => ({ description }));
	}
	onNoteChange = (e) => {
		const note = e.target.value;
		this.setState(() => ({ note }));
	}
	onAmountChange = (e) => {
		const amount = e.target.value;
		// Me corrigir REGEX
		if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
			this.setState(() => ({ amount }));
		}
	}
	onDataChange = (createdAt) => {
		if (createdAt) {
			this.setState(() => ({  createdAt }));
		}
	}
	onFocusChange = ({ focused }) => {
		this.setState(() => ({  calendarFocused: focused }));
	}
	onSubmit = (e) => {
		e.preventDefault();

		if (!this.state.description || !this.state.amount) {
			this.setState(() => ({ error: 'Please provide description and amount' }));
		} else {
			this.setState(() => ({ error: '' }));
			this.props.onSubmit({
				description: this.state.description,
				amount: parseFloat(this.state.amount),
				createdAt: this.state.createdAt.valueOf(),
				note: this.state.note
			})
		}
	}
	render() {
		return (
			<div>
			{this.state.error && <p>{this.state.error}</p>}
			<form onSubmit={this.onSubmit}>
				<input
				type="text"
				placeholder="Description"
				autoFocus
				value={this.state.description}
				onChange={this.onDescriptionChange}
				/>
				<input
				type="text"
				placeholder="Amount"
				onChange={this.onAmountChange}
				/>
				<SingleDatePicker 
					date={this.state.createdAt}
					onDateChange={this.onDataChange}
					focused={this.state.calendarFocused}
					onFocusChange={this.onFocusChange}
					numberOfMonths={1}
					isOutsideRange={() => false}
				/>
				<textarea
				placeholder="Add a note for your expense (optional)"
				value={this.state.note}
				onChange={this.onNoteChange}
				/>
				<button>Add Expense</button>
			</form>
			</div>
		)
	}
}