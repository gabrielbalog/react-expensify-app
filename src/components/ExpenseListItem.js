import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

export const ExpenseListItem = ({ id, description, amount, createdAt, dispatch }) => (
	<div>
		<Link to={`/edit/${id}`}>
			<h3>{description}</h3>
		</Link>
		<p>{amount} - {createdAt}</p>
	</div>
);



export default connect()(ExpenseListItem);