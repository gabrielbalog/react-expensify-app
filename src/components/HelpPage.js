import React from 'react';

const HelpPage = () => (
	<div>
		This is from help Page
	</div>
);

export default HelpPage;