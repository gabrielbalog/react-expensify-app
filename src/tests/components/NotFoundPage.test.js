import React from 'react';
import { shallow } from 'enzyme';
import { NotFoundPage } from '../../components/NotFoundPage';
import expenses from '../fixtures/expenses';

test('should render NotFoundPage with one expense', () => {
	const wrapper = shallow(<NotFoundPage />);
	expect(wrapper).toMatchSnapshot();
});