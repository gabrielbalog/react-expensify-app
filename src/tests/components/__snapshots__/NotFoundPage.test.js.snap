// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NotFoundPage with one expense 1`] = `
<div>
  404 - 
  <Link
    replace={false}
    to="/"
  >
    Go Home
  </Link>
</div>
`;
